src/storage/src/render/upsert.rs

```diff
@@ -106,11 +125,13 @@ impl<H: Digest> Hasher for DigestHasher<H> {
 /// Resumes an upsert computation at `resume_upper` given as inputs a collection of upsert commands
 /// and the collection of the previous output of this operator.
 pub(crate) fn upsert<G: Scope, O: timely::ExchangeData + Ord>(
-    input: &Collection<G, (UpsertKey, Option<Result<Row, UpsertError>>, O), Diff>,
+    input: &Collection<G, (UpsertKey, Option<UpsertValue>, O), Diff>,
     mut key_indices: Vec<usize>,
     resume_upper: Antichain<G::Timestamp>,
     previous: Collection<G, Result<Row, DataflowError>, Diff>,
     previous_token: Option<Rc<dyn Any>>,
+    source_id: GlobalId,
+    storage_state: &crate::storage_state::StorageState,
```

Comment:
can you pass the config `RawSourceCreationConfig` object here instead? It contains the base metrics, source id, and timely configuration info you need. We should at some point rename that object as it has useful information for all operators for the ingestion